ttnn/cpp/ttnn/operations/conv2d/conv2d_pybind.hpp

```diff
@@ -55,6 +55,65 @@ void py_module(py::module& module) {
         py::arg("bias_tensor") = std::nullopt,
         py::arg("conv_config") = std::nullopt);
 
+    module.def(
+        "optimized_conv",
+        &optimized_conv,
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg("bias").noconvert() = std::nullopt,
+        py::arg("conv_reader_indices").noconvert() = std::nullopt,
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert(),
+        py::arg().noconvert() = 0,
+        py::arg("output_mem_config").noconvert() = std::nullopt,
+        py::arg("output_dtype").noconvert() = std::nullopt,
+        py::arg("input_tensor_shape").noconvert() = std::nullopt,
```

Comment:
why does it need to be an argument and can't be retrieved from the tensor itself?

Reply:
In convolution, the 4D input tensor is reshaped into a 2D matrix, and so the original input tensor is lost. The user has to manually pass this as well.

Reply:
yes, this is specific to these ops where input is expected to be flattened -- which makes original shape to be lost. The new tensor API which is WIP will replace current Tensor infra and will store all necessary metadata. Then we can get rid of this.